test/passing/let_binding_compact.ml

```diff
@@ -0,0 +1,21 @@
+let z = this one is pretty looooooooooooooooooooooooooooooooooong
+let z = so is this oooooooooooooooooooooooooooooooooooooooooooone
+
+let f x = x + 1
+
+let z = this one is pretty looooooooooooooooooooooooooooooooooong
+let z = so is this oooooooooooooooooooooooooooooooooooooooooooone
+let g = ()
+
+let f = function
+  | `a | `b | `c -> foo
+  | `xxxxxxxxxxxxxxxxxx ->
+      yyyyyyyyyyyyyyyyyyyyyyyy
+        zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz
+        zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz
+        zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz
+
+let x = 1
+let y = 2
+let z = this one is pretty looooooooooooooooooooooooooooooooooong
+let z = so is this oooooooooooooooooooooooooooooooooooooooooooone
```

Comment:
It would be good to add to this test so that it covers the interaction between different forms of structure items.